### PR TITLE
doc: lwm2m_client: Update documentation

### PR DESCRIPTION
**Pull request by juhaylinen**
Add link to Zephyr LwM2M sample which is demonstrates basic LwM2M functionality.
Update some configuration values.

### CODE REVIEW

**Review by SeppoTakalo — APPROVED** [review]

File: samples/cellular/lwm2m_client/README.rst [hunk 1]
```diff
@@ -8,7 +8,9 @@ Cellular: LwM2M Client
    :depth: 2
 
 The LwM2M Client demonstrates usage of the :term:`Lightweight Machine to Machine (LwM2M)` protocol to connect a Thingy:91 or an nRF91 Series DK to an LwM2M server through LTE.
-This sample uses the :ref:`lib_lwm2m_client_utils` library.
+This sample uses the :ref:`lib_lwm2m_client_utils` library and the :ref:`lib_lwm2m_location_assistance` library.
```
Comment (peknis) [inline samples/cellular/lwm2m_client/README.rst:11]:
```suggestion
This sample uses the :ref:`lib_lwm2m_client_utils` and :ref:`lib_lwm2m_location_assistance` libraries.
```

File: samples/cellular/lwm2m_client/README.rst [hunk 2]
```diff
@@ -8,7 +8,9 @@ Cellular: LwM2M Client
...
+
+The sample has implementation for many cellular specific features such as location services, firmware update and power saving optimizations. For network agnostic and basic LwM2M client sample see: :zephyr:code-sample:`lwm2m-client`.
```
Comment (peknis) [inline samples/cellular/lwm2m_client/README.rst:13]:
```suggestion
The sample has implementation for many cellular-specific features, such as location services, firmware update, and power saving optimizations. For a network-agnostic and basic LwM2M client sample, see :zephyr:code-sample:`lwm2m-client`.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

File: samples/cellular/lwm2m_client/README.rst [hunk 3]
```diff
@@ -8,7 +8,9 @@ Cellular: LwM2M Client
    :depth: 2
 
 The LwM2M Client demonstrates usage of the :term:`Lightweight Machine to Machine (LwM2M)` protocol to connect a Thingy:91 or an nRF91 Series DK to an LwM2M server through LTE.
-This sample uses the :ref:`lib_lwm2m_client_utils` library.
+This sample uses the :ref:`lib_lwm2m_client_utils` and :ref:`lib_lwm2m_location_assistance` libraries.
+
+The sample has implementation for many cellular-specific features, such as location services, firmware update, and power saving optimizations. For a network-agnostic and basic LwM2M client sample, see :zephyr:code-sample:`lwm2m-client`.
```
Comment (peknis) [inline samples/cellular/lwm2m_client/README.rst:13]:
```suggestion
The sample has implementation for many cellular-specific features, such as location services, firmware update, and power saving optimizations.
For a network-agnostic and basic LwM2M client sample, see :zephyr:code-sample:`lwm2m-client`.
```

**Review by peknis — APPROVED** [review]
One more minor nit before you ask for merge.

**Pull request merged**